scripts/library.sh

```diff
@@ -437,3 +428,59 @@ function lint_markdown() {
   # https://github.com/markdownlint/markdownlint
   run_lint_tool mdl "linting markdown files" "-r ~MD013" $@
 }
+
+# Call a function and verify its return value and output.
+# Parameters: $1 - expected return code.
+#             $2 - expected output ("" if no output is expected)
+#             $3 ..$n - function to call and its parameters.
+function test_function() {
+  local expected_retcode=$1
+  local expected_string=$2
+  local output="$(mktemp)"
+  local output_code="$(mktemp)"
+  shift 2
+  echo -n "$(trap '{ echo $? > ${output_code}; }' EXIT ; "$@")" &> ${output}
+  local retcode=$(cat ${output_code})
+  if [[ ${retcode} -ne ${expected_retcode} ]]; then
+    cat ${output}
+    echo "Return code ${retcode} doesn't match expected return code ${expected_retcode}"
+    return 1
+  fi
+  if [[ -n "${expected_string}" ]]; then
+    local found=1
+    grep "${expected_string}" ${output} > /dev/null || found=0
+    if (( ! found )); then
+      cat ${output}
+      echo "String '${expected_string}' not found"
+      return 1
+    fi
+  else
+    if [[ -s ${output} ]]; then
+      ls ${output}
+      cat ${output}
+      echo "Unexpected output"
+      return 1
+    fi
+  fi
+  echo "'$@' returns code ${expected_retcode} and displays '${expected_string}'"
+}
+
+# Mock gcloud function for testing purpose
+function mock_gcloud_function() {
+  set -e
+  function gcloud() {
+    echo ""
+  }
+  "$@" 2>&1
+}
+
+function check_is_int() {
+  [[ -n $1 && $1 =~ ^[0-9]+$ ]] && return 0
+  return 1
+}
+
+# Safety checks, make sure not release/nightly gcr
+function check_is_protected_gcr() {
```

Comment:
Rename to `is_protected_gcr`, it will make it more readable in conditionals.

Reply:
done, please review